bootstrap/eks/api/v1alpha3/condition_consts.go

```diff
@@ -0,0 +1,30 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha3
+
+import clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha3"
+
+// Conditions and condition Reasons for the EKSConfig object
+
+const (
+	// DataSecretAvailableCondition documents the status of the bootstrap secret generation process.
+	//
+	// NOTE: When the DataSecret generation starts the process completes immediately and within the
+	// same reconciliation, so the user will always see a transition from Wait to Generated without having
+	// evidence that BootstrapSecret generation is started/in progress.
+	DataSecretAvailableCondition clusterv1.ConditionType = "DataSecretAvailable"
+)
```

Comment:
Many more conditions here: https://github.com/kubernetes-sigs/cluster-api/blob/master/bootstrap/kubeadm/api/v1alpha3/condition_consts.go

Reply:
cc @NicoleYson I think was working on the condition stuffs

Reply:
https://github.com/newrelic-forks/cluster-api-provider-aws/pull/13/files#diff-598f3bde64955c8ab3c824a50f6d8c6eR29 
😂 I saw that! I only pulled in what I needed so far, I still have a bit to figure out.